realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -766,7 +844,13 @@ void BaseRealSenseNode::publishAlignedDepthToOthers(rs2::frameset frames, const
                          _depth_aligned_info_publisher,
                          _depth_aligned_image_publishers, _depth_aligned_seq,
                          _depth_aligned_camera_info, _optical_frame_id,
-                         _depth_aligned_encoding);
+                         _depth_aligned_encoding,
+                         true, true);
+        }
+        else
+        {
+            clearMonitoredTopic(sip, TOPIC_ALIGNED_INFO);
+            clearMonitoredTopic(sip, TOPIC_ALIGNED_IMAGE);
```

Comment:
Does this mean until we have subscribers, we wont be checking the topic until then? I think it might be desirable (and IMO makes more sense since we have a configurable to determine the timeout action) to perform the pre-configured `_timeout_action` as soon as the topic is detected not being published

Reply:
Yeah, since the publishing on these topics is conditioned on there being subscribers to the topics, I reset the timestamp for topics that have no subscribers to uninitialized.  This way monitoring only happens to actively subscribed topics and the node doesn't reset or shutdown itself just because someone subscribed and then unsubscribed to one of the topics.

Reply:
But also, yes, no monitoring/action is actually performed until a message is actually published on a given topic, so if the driver starts up "correctly", but for some reason fails to publish any data even though it's supposed to, the timeout action won't occur. 

That case is probably better handled with some external monitor with a larger timeout.

Reply:
> This way monitoring only happens to actively subscribed topics and the node doesn't reset or shutdown itself just because someone subscribed and then unsubscribed to one of the topics.

Ah, that makes sense. +1